packages/@aws-cdk/aws-eks/lib/k8s-cluster-autoscaler.ts

```diff
@@ -0,0 +1,389 @@
+import autoscaling = require('@aws-cdk/aws-autoscaling');
+import iam = require('@aws-cdk/aws-iam');
+import core = require('@aws-cdk/core');
+import cdk = require('@aws-cdk/core');
+import { Cluster } from './cluster';
+import { KubernetesResource } from './k8s-resource';
+
+/**
+ * A Cluster Autoscaler
+ */
+export interface IKubernetesClusterAutoscaler extends core.IConstruct {
+  /**
+   *  The Cluster Autoscalers Cluster.
+   *
+   * [disable-awslint:ref-via-interface]
+   */
+  readonly cluster: Cluster;
+
+  /**
+   *  The IAM policy created by this construct.
+   */
+  readonly policy: iam.IPolicy;
+
+  /**
+   * The Kubernetes Resource that defines the Cluster Autoscaler K8s resources.
+   */
+  readonly clusterAutoscaler: KubernetesResource;
+}
+
+/**
+ * The properties for the Cluster Autoscaler.
+ */
+export interface KubernetesClusterAutoscalerProps {
+
+  /**
+   * The EKS cluster to deploy the cluster autoscaler to.
+   *
+   * [disable-awslint:ref-via-interface]
+   */
+  readonly cluster: Cluster;
+
+  /**
+   * An array of Autoscaling Groups, known as node groups, to configure for autoscaling.
+   *
+   * [disable-awslint:ref-via-interface]
```

Comment:
why? Can't we accept `IAutoScalingGroup` here?

Reply:
`Tag.add` only accepts the type of `Construct`. `IAutoScalingGroup` was throwing errors.

Reply:
Got it, and I guess these tags are required... You could technically only add tags if `nodeGroups instanceof AutoScalingGroup` (i.e. it's an "owned" ASG that you can actually set the tags on), but I guess this is fine. We can always relax.